tests/Unit/PointwiseFunctions/AnalyticSolutions/Elasticity/HalfSpaceMirror.py

```diff
@@ -0,0 +1,141 @@
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+import numpy as np
+import scipy.integrate as integrate
+from scipy.special import jv
+from Elasticity.ConstitutiveRelations.IsotropicHomogeneous import (
+    lame_parameter)
+
+
+def alpha(k, beam_width, applied_force):
+    return 1 / (2. * np.pi) * np.exp(-(k * beam_width / 2.)**2) * applied_force
+
+
+def integrand_xi_w(k, w, z, lame_parameter, shear_modulus, beam_width,
+                   applied_force):
+    return 1 / (2. * shear_modulus) * jv(1, k * w) * np.exp(-k * z) * \
+        (1 - (lame_parameter + 2. * shear_modulus) / \
+        (lame_parameter + shear_modulus) + k * z) * \
+        alpha(k, beam_width, applied_force)
+
+
+def integrand_xi_z(k, w, z, lame_parameter, shear_modulus, beam_width,
+                   applied_force):
+    return 1 / (2. * shear_modulus) * jv(0, k * w) * np.exp(-k * z) * \
+        (1 + shear_modulus / (lame_parameter + shear_modulus) + k * z) * \
+        alpha(k, beam_width, applied_force)
+
+
+def integrand_theta(k, w, z, lame_parameter, shear_modulus, beam_width,
+                    applied_force):
+    return 1 / (2. * shear_modulus) * k * jv(0, k * w) * np.exp(-k * z) * \
+        (-2. * shear_modulus /(lame_parameter + shear_modulus)) * \
+        alpha(k, beam_width, applied_force)
+
+
+def integrand_strain_rz(k, w, z, lame_parameter, shear_modulus, beam_width,
+                        applied_force):
+    return -1 / (2. * shear_modulus) * k * jv(1, k * w) * (k * z) * \
+        np.exp(-k * z) * alpha(k, beam_width, applied_force)
+
+
+def integrand_strain_zz(k, w, z, lame_parameter, shear_modulus, beam_width,
+                        applied_force):
+    return 1 / (2. * shear_modulus) * k * jv(0, k * w) * np.exp(-k * z) * \
+        (-shear_modulus /(lame_parameter + shear_modulus) - k * z) * \
+        alpha(k, beam_width, applied_force)
+
+
+def displacement(r, beam_width, applied_force, bulk_modulus, shear_modulus):
+    local_lame_parameter = lame_parameter(bulk_modulus, shear_modulus)
+    x = r[0]
+    y = r[1]
+    z = r[2]
+    w = np.sqrt(x**2 + y**2)
+    xi_w = integrate.quad(
+        lambda k: integrand_xi_w(k, w, z, local_lame_parameter, shear_modulus,
+                                 beam_width, applied_force),
+        0,
+        np.inf,
+        limit=100,
+        epsabs=1e-13)[0]
+
+    xi_z = integrate.quad(
+        lambda k: integrand_xi_z(k, w, z, local_lame_parameter, shear_modulus,
+                                 beam_width, applied_force),
+        0,
+        np.inf,
+        limit=100,
+        epsabs=1e-13)[0]
+
+    if w == 0.:
```

Comment:
Is this because `HalfSpaceMirror.py` is used only here and only for testing purposes, so that there's never an instance where the functions are fed an array?

Reply:
Nevermind. If nothing but the integration wouldn't work with an array. Sorry.